### ISSUE TITLE
Alacritty randomly closes

### ISSUE THREAD

**Issue by juchiast** [jwilm/alacritty#2907]
Which operating system does the issue occur on? Linux, Fedora 30.

If on linux, are you using X11 or Wayland? X11, i3.

I'm using latest master branch. Mostly while using Vim, Alacritty randomly crashes.

**Comment by kchibisov** [comment]
Please provide backtrace for a crash. You can use `coredumpctl gdb alacritty`, it should give you the most recent crash.

**Comment by juchiast** [comment]
@kchibisov Just checked again, it didn't crash. Here's the log:

```
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=41, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=42, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=43, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=44, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=45, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=46, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=47, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1955] Setting attribute: Reset
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=48, col=78
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1534] Moving forward: 11
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=2, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:2029] Unsetting mode: BlinkingCursor
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:2063] ... unimplemented mode
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1988] Setting mode: ShowCursor
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:2029] Unsetting mode: ShowCursor
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=48, col=78
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=3, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:2029] Unsetting mode: BlinkingCursor
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:2063] ... unimplemented mode
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1988] Setting mode: ShowCursor
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:2029] Unsetting mode: ShowCursor
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=48, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1803] Clearing line: Right
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1467] Going to: line=48, col=0
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:2029] Unsetting mode: BlinkingCursor
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:2063] ... unimplemented mode
[2019-10-23 21:00] [TRACE] [alacritty_terminal/src/term/mod.rs:1988] Setting mode: ShowCursor
[2019-10-23 21:00] [INFO] Goodbye
```

More on reproducing the error: When I get myself into #2906, pressing "w" will cause the terminal to close.

**Comment by kchibisov** [comment]
Could you repro with `alacritty --config-file /dev/null`?

**Comment by juchiast** [comment]
> Could you repro with alacritty --config-file /dev/null?

Probably not, using it for a while and still not see the bug. Sometime arrow keys are not usable, but it doesn't close.

**Comment by kchibisov** [comment]
Would you mind to share your config then?

**Comment by juchiast** [comment]
Here:
```
window:
  dimensions:
    columns: 80
    lines: 24

  # Window padding
  padding:
    x: 2
    y: 2

  # Window decorations
  decorations: none

tabspaces: 4

scrolling:
  # How many lines of scrollback to keep,
  # '0' will disable scrolling.
  history: 10000

  # Number of lines the viewport will move for every line
  # scrolled when scrollback is enabled (history > 0).
  multiplier: 3

  # Automatically scroll to the bottom when new text is written
  # to the terminal.
  auto_scroll: false

draw_bold_text_with_bright_colors: true

# Font configuration (changes require restart)
#
# Important font attributes like antialiasing, subpixel aa, and hinting can be
# controlled through fontconfig. Specifically, the following attributes should
# have an effect:
#
# * hintstyle
# * antialias
# * lcdfilter
# * rgba
#
# For instance, if you wish to disable subpixel antialiasing, you might set the
# rgba property to "none". If you wish to completely disable antialiasing, you
# can set antialias to false.
#
# Please see these resources for more information on how to use fontconfig
#
# * https://wiki.archlinux.org/index.php/font_configuration#Fontconfig_configuration
# * file:///usr/share/doc/fontconfig/fontconfig-user.html
font:
  normal:
    family: Source Code Pro
    style: Medium

  bold:
    family: Source Code Pro
    style: Bold

  italic:
    family: Source Code Pro
    style: Italic

  size: 9

  # Offset is the extra space around each character. offset.y can be thought of
  # as modifying the linespacing, and offset.x as modifying the letter spacing.
  offset:
    x: 0
    y: 0

  # Glyph offset determines the locations of the glyphs within their cells with
  # the default being at the bottom. Increase the x offset to move the glyph to
  # the right, increase the y offset to move the glyph upward.
  glyph_offset:
    x: 0
    y: 0

# Should display the render timer
# render_timer: false

# Colors (Dracula)
colors:
  # Default colors
  primary:
    background: '0x282a36'
    foreground: '0xf8f8f2'

  # Normal colors
  normal:
    black:   '0x000000'
    red:     '0xff5555'
    green:   '0x50fa7b'
    yellow:  '0xf1fa8c'
    blue:    '0xbd93f9'
    magenta: '0xff79c6'
    cyan:    '0x8be9fd'
    white:   '0xbfbfbf'

  # Bright colors
  bright:
    black:   '0x4d4d4d'
    red:     '0xff6e67'
    green:   '0x5af78e'
    yellow:  '0xf4f99d'
    blue:    '0xcaa9fa'
    magenta: '0xff92d0'
    cyan:    '0x9aedfe'
    white:   '0xe6e6e6'

# Visual Bell
#
# Any time the BEL code is received, Alacritty "rings" the visual bell. Once
# rung, the terminal background will be set to white and transition back to the
# default background color. You can control the rate of this transition by
# setting the `duration` property (represented in milliseconds). You can also
# configure the transition function by setting the `animation` property.
#
# Possible values for `animation`
# `Ease`
# `EaseOut`
# `EaseOutSine`
# `EaseOutQuad`
# `EaseOutCubic`
# `EaseOutQuart`
# `EaseOutQuint`
# `EaseOutExpo`
# `EaseOutCirc`
# `Linear`
#
# To completely disable the visual bell, set its duration to 0.
#
visual_bell:
  animation: EaseOutExpo
  duration: 0

# Background opacity
background_opacity: 0.85

# Mouse bindings
#
# Currently doesn't support modifiers. Both the `mouse` and `action` fields must
# be specified.
#
# Values for `mouse`:
# - Middle
# - Left
# - Right
# - Numeric identifier such as `5`
#
# Values for `action`:
# - Paste
# - PasteSelection
# - Copy (TODO)
mouse_bindings:
  - { mouse: Middle, action: PasteSelection }

mouse:
  hide_when_typing: false
  # Click settings
  #
  # The `double_click` and `triple_click` settings control the time
  # alacritty should wait for accepting multiple clicks as one double
  # or triple click.
  double_click: { threshold: 300 }
  triple_click: { threshold: 300 }

  # Faux Scrollback
  #
  # The `faux_scrollback_lines` setting controls the number
  # of lines the terminal should scroll when the alternate
  # screen buffer is active. This is used to allow mouse
  # scrolling for applications like `man`.
  #
  # To disable this completely, set `faux_scrollback_lines` to 0.
  faux_scrollback: 2

  url:
    # URL launcher
    #
    # This program is executed when clicking on a text which is recognized as a URL.
    # The URL is always added to the command as the last parameter.
    launcher: xdg-open
    modifiers: Control

selection:
  semantic_escape_chars: ",│`|:\"' ()[]{}<>"

dynamic_title: true

# Style of the cursor
#
# Values for 'cursor_style':
# - Block
# - Underline
# - Beam
cursor:
  style: Block

# Live config reload (changes require restart)
live_config_reload: false

# Shell
#
# You can set shell.program to the path of your favorite shell, e.g. /bin/fish.
# Entries in shell.args are passed unmodified as arguments to the shell.
#
# shell:
#   program: /bin/bash
#   args:
#     - --login

# Key bindings
#
# Each binding is defined as an object with some properties. Most of the
# properties are optional. All of the alphabetical keys should have a letter for
# the `key` value such as `V`. Function keys are probably what you would expect
# as well (F1, F2, ..). The number keys above the main keyboard are encoded as
# `Key1`, `Key2`, etc. Keys on the number pad are encoded `Number1`, `Number2`,
# etc.  These all match the glutin::VirtualKeyCode variants.
#
# A list with all available `key` names can be found here:
# https://docs.rs/glutin/*/glutin/enum.VirtualKeyCode.html#variants
#
# Possible values for `mods`
# `Command`, `Super` refer to the super/command/windows key
# `Control` for the control key
# `Shift` for the Shift key
# `Alt` and `Option` refer to alt/option
#
# mods may be combined with a `|`. For example, requiring control and shift
# looks like:
#
# mods: Control|Shift
#
# The parser is currently quite sensitive to whitespace and capitalization -
# capitalization must match exactly, and piped items must not have whitespace
# around them.
#
# Either an `action`, `chars`, or `command` field must be present.
#   `action` must be one of `Paste`, `PasteSelection`, `Copy`, or `Quit`.
#   `chars` writes the specified string every time that binding is activated.
#     These should generally be escape sequences, but they can be configured to
#     send arbitrary strings of bytes.
#   `command` must be a map containing a `program` string, and `args` array of
#     strings. For example:
#     - { ... , command: { program: "alacritty", args: ["-e", "vttest"] } }
#
# Want to add a binding (e.g. "PageUp") but are unsure what the X sequence
# (e.g. "\x1b[5~") is? Open another terminal (like xterm) without tmux,
# then run `showkey -a` to get the sequence associated to a key combination.
key_bindings:
  - { key: V,        mods: Control|Shift,    action: Paste               }
  - { key: C,        mods: Control|Shift,    action: Copy                }

  - { key: PageUp,   mods: Control,   action: ScrollPageUp               }
  - { key: PageDown, mods: Control,   action: ScrollPageDown             }
  - { key: End,      mods: Control,   action: ScrollToBottom             }
  - { key: Home,     mods: Control,   action: ScrollToTop                }
  - { key: K,        mods: Command,   action: ClearHistory               }

  - { key: Q,        mods: Command, action: Quit                         }
  - { key: W,        mods: Command, action: Quit                         }
  - { key: Insert,   mods: Shift,   action: PasteSelection               }
  - { key: Key0,     mods: Control, action: ResetFontSize                }
  - { key: Equals,   mods: Control, action: IncreaseFontSize             }
  - { key: Subtract, mods: Control, action: DecreaseFontSize             }
  - { key: Home,                    chars: "\x1bOH",   mode: AppCursor   }
  - { key: Home,                    chars: "\x1b[H",   mode: ~AppCursor  }
  - { key: End,                     chars: "\x1bOF",   mode: AppCursor   }
  - { key: End,                     chars: "\x1b[F",   mode: ~AppCursor  }
  - { key: PageUp,   mods: Shift,   chars: "\x1b[5;2~"                   }
  # - { key: PageUp,   mods: Control, chars: "\x1b[5;5~"                   }
  - { key: PageUp,                  chars: "\x1b[5~"                     }
  - { key: PageDown, mods: Shift,   chars: "\x1b[6;2~"                   }
  # - { key: PageDown, mods: Control, chars: "\x1b[6;5~"                   }
  - { key: PageDown,                chars: "\x1b[6~"                     }
  - { key: Tab,      mods: Shift,   chars: "\x1b[Z"                      }
  - { key: Back,                    chars: "\x7f"                        }
  - { key: Back,     mods: Alt,     chars: "\x1b\x7f"                    }
  - { key: Insert,                  chars: "\x1b[2~"                     }
  - { key: Delete,                  chars: "\x1b[3~"                     }
  - { key: Left,     mods: Shift,   chars: "\x1b[1;2D"                   }
  - { key: Left,     mods: Control, chars: "\x1b[1;5D"                   }
  - { key: Left,     mods: Alt,     chars: "\x1b[1;3D"                   }
  - { key: Left,                    chars: "\x1b[D",   mode: ~AppCursor  }
  - { key: Left,                    chars: "\x1bOD",   mode: AppCursor   }
  - { key: Right,    mods: Shift,   chars: "\x1b[1;2C"                   }
  - { key: Right,    mods: Control, chars: "\x1b[1;5C"                   }
  - { key: Right,    mods: Alt,     chars: "\x1b[1;3C"                   }
  - { key: Right,                   chars: "\x1b[C",   mode: ~AppCursor  }
  - { key: Right,                   chars: "\x1bOC",   mode: AppCursor   }
  - { key: Up,       mods: Shift,   chars: "\x1b[1;2A"                   }
  - { key: Up,       mods: Control, chars: "\x1b[1;5A"                   }
  - { key: Up,       mods: Alt,     chars: "\x1b[1;3A"                   }
  - { key: Up,                      chars: "\x1b[A",   mode: ~AppCursor  }
  - { key: Up,                      chars: "\x1bOA",   mode: AppCursor   }
  - { key: Down,     mods: Shift,   chars: "\x1b[1;2B"                   }
  - { key: Down,     mods: Control, chars: "\x1b[1;5B"                   }
  - { key: Down,     mods: Alt,     chars: "\x1b[1;3B"                   }
  - { key: Down,                    chars: "\x1b[B",   mode: ~AppCursor  }
  - { key: Down,                    chars: "\x1bOB",   mode: AppCursor   }
  - { key: F1,                      chars: "\x1bOP"                      }
  - { key: F2,                      chars: "\x1bOQ"                      }
  - { key: F3,                      chars: "\x1bOR"                      }
  - { key: F4,                      chars: "\x1bOS"                      }
  - { key: F5,                      chars: "\x1b[15~"                    }
  - { key: F6,                      chars: "\x1b[17~"                    }
  - { key: F7,                      chars: "\x1b[18~"                    }
  - { key: F8,                      chars: "\x1b[19~"                    }
  - { key: F9,                      chars: "\x1b[20~"                    }
  - { key: F10,                     chars: "\x1b[21~"                    }
  - { key: F11,                     chars: "\x1b[23~"                    }
  - { key: F12,                     chars: "\x1b[24~"                    }
  - { key: F1,       mods: Shift,   chars: "\x1b[1;2P"                   }
  - { key: F2,       mods: Shift,   chars: "\x1b[1;2Q"                   }
  - { key: F3,       mods: Shift,   chars: "\x1b[1;2R"                   }
  - { key: F4,       mods: Shift,   chars: "\x1b[1;2S"                   }
  - { key: F5,       mods: Shift,   chars: "\x1b[15;2~"                  }
  - { key: F6,       mods: Shift,   chars: "\x1b[17;2~"                  }
  - { key: F7,       mods: Shift,   chars: "\x1b[18;2~"                  }
  - { key: F8,       mods: Shift,   chars: "\x1b[19;2~"                  }
  - { key: F9,       mods: Shift,   chars: "\x1b[20;2~"                  }
  - { key: F10,      mods: Shift,   chars: "\x1b[21;2~"                  }
  - { key: F11,      mods: Shift,   chars: "\x1b[23;2~"                  }
  - { key: F12,      mods: Shift,   chars: "\x1b[24;2~"                  }
  - { key: F1,       mods: Control, chars: "\x1b[1;5P"                   }
  - { key: F2,       mods: Control, chars: "\x1b[1;5Q"                   }
  - { key: F3,       mods: Control, chars: "\x1b[1;5R"                   }
  - { key: F4,       mods: Control, chars: "\x1b[1;5S"                   }
  - { key: F5,       mods: Control, chars: "\x1b[15;5~"                  }
  - { key: F6,       mods: Control, chars: "\x1b[17;5~"                  }
  - { key: F7,       mods: Control, chars: "\x1b[18;5~"                  }
  - { key: F8,       mods: Control, chars: "\x1b[19;5~"                  }
  - { key: F9,       mods: Control, chars: "\x1b[20;5~"                  }
  - { key: F10,      mods: Control, chars: "\x1b[21;5~"                  }
  - { key: F11,      mods: Control, chars: "\x1b[23;5~"                  }
  - { key: F12,      mods: Control, chars: "\x1b[24;5~"                  }
  - { key: F1,       mods: Alt,     chars: "\x1b[1;6P"                   }
  - { key: F2,       mods: Alt,     chars: "\x1b[1;6Q"                   }
  - { key: F3,       mods: Alt,     chars: "\x1b[1;6R"                   }
  - { key: F4,       mods: Alt,     chars: "\x1b[1;6S"                   }
  - { key: F5,       mods: Alt,     chars: "\x1b[15;6~"                  }
  - { key: F6,       mods: Alt,     chars: "\x1b[17;6~"                  }
  - { key: F7,       mods: Alt,     chars: "\x1b[18;6~"                  }
  - { key: F8,       mods: Alt,     chars: "\x1b[19;6~"                  }
  - { key: F9,       mods: Alt,     chars: "\x1b[20;6~"                  }
  - { key: F10,      mods: Alt,     chars: "\x1b[21;6~"                  }
  - { key: F11,      mods: Alt,     chars: "\x1b[23;6~"                  }
  - { key: F12,      mods: Alt,     chars: "\x1b[24;6~"                  }
  - { key: F1,       mods: Super,   chars: "\x1b[1;3P"                   }
  - { key: F2,       mods: Super,   chars: "\x1b[1;3Q"                   }
  - { key: F3,       mods: Super,   chars: "\x1b[1;3R"                   }
  - { key: F4,       mods: Super,   chars: "\x1b[1;3S"                   }
  - { key: F5,       mods: Super,   chars: "\x1b[15;3~"                  }
  - { key: F6,       mods: Super,   chars: "\x1b[17;3~"                  }
  - { key: F7,       mods: Super,   chars: "\x1b[18;3~"                  }
  - { key: F8,       mods: Super,   chars: "\x1b[19;3~"                  }
  - { key: F9,       mods: Super,   chars: "\x1b[20;3~"                  }
  - { key: F10,      mods: Super,   chars: "\x1b[21;3~"                  }
  - { key: F11,      mods: Super,   chars: "\x1b[23;3~"                  }
  - { key: F12,      mods: Super,   chars: "\x1b[24;3~"                  }
```

**Comment by kchibisov** [comment]
Have you changed terminfo to match your tabspace value? Change it back to default otherwise and retest.

**Comment by kchibisov** [comment]
Maybe you have some problem with Command(Win), since you have a binding for `K` and `W` and you've mentioned in other issue(iirc), that your `K` also doesn't work in vim some time.

**Comment by juchiast** [comment]
> Have you changed terminfo to match your tabspace value?

I don't understand this, so probably I didn't to it.

**Comment by juchiast** [comment]
I use Win+K to switch between windows (i3wm), now I can reproduce this by pressing it really fast.

**Comment by juchiast** [comment]
Closed as duplication of #2906
